internal/parser/rfc5322_address_list.go

```diff
@@ -0,0 +1,39 @@
+package parser
+
+/*
+#include "src/rfc5322/rfc5322_parser_capi.h"
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"net/mail"
+	"unsafe"
+)
+
+func ParseRFC5322AddressList(input string) ([]*mail.Address, error) {
+	parser := C.RFC5322AddressList_new()
+	defer C.RFC5322AddressList_free(parser)
+
+	cstr := C.CString(input)
+	defer C.free(unsafe.Pointer(cstr))
```

Comment:
i like how we avoided swig here, do you think that's something we could realistically do with the IMAP parser too? i used swig because the IMAP parser expects C++ types as input, but i guess we could rewrite them into pure C types

Reply:
I will do a pass over the IMAP parser once I'm done with this and I'll see what I can do from there.